recipes/innoextract/all/conanfile.py

```diff
@@ -0,0 +1,71 @@
+import os
+import shutil
+from conans import ConanFile, CMake, tools
+
+required_conan_version = ">=1.33.0"
+
+
+class InnoextractConan(ConanFile):
+    name = "innoextract"
+    description = "Extract contents of Inno Setup installers"
+    license = "innoextract License"
+    topics = ("inno-setup", "decompression")
+    homepage = "https://constexpr.org/innoextract/"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    requires = (
+        ("boost/1.78.0", "private"),
+        ("xz_utils/5.2.5", "private"),
+        ("libiconv/1.16", "private")
```

Comment:
```suggestion
        ("boost/1.78.0", "private"),
        ("xz_utils/5.2.5", "private"),
        ("libiconv/1.16", "private")
```

And what happen if those libs are shared? private keyword in conan is fragile, don't use it except you know all the consequences.

Reply:
This is a tool used as a build_requirement, not a library. If they aren't private, then even if the innoextract binary exists, Conan is trying to supply the libraries, even going so far as to waste time building them.

It probably won't build if those libraries are shared, but I could enforce that by setting the shared options for the requirements in `default_options`, no?

Reply:
I know, but it can be fragile. There are still issues in option handling by conan between build & host profile, and forcing shared option of dependencies in a build requirement may have side effects in requirements graph.

Reply:
I'll try taking it the private keyword, then. If it really causes problems for us, I'll put in our fork.

Reply:
I'm making the changes to not make the requirements private, but I run into this:
```
Build requirements packages
    boost/1.78.0:3f33955e7918ca665eba360boost/1.78.0: WARN: Can't find a 'boost/1.78.0' package for the specified settings, options and dependencies:
- Settings: arch=x86_64, build_type=Release, compiler=Visual Studio, compiler.runtime=MD, compiler.version=15, os=Windows
- Options: asio_no_deprecated=False, buildid=None, bzip2=True, debug_level=0, diagnostic_definitions=False, error_code_header_only=False, extra_b2_flags=--hash, filesystem_no_deprecated=False, header_only=False, i18n_backend=deprecated, i18n_backend_iconv=off, i18n_backend_icu=False, layout=system, lzma=False, magic_autolink=False, multithreading=True, namespace=boost, namespace_alias=False, numa=True, pch=False, python_executable=None, python_version=None, segmented_stacks=False, shared=False, system_no_deprecated=False, visibility=hidden, without_atomic=False, without_chrono=False, without_container=False, without_context=False, without_contract=False, without_coroutine=False, without_date_time=False, without_exception=False, without_fiber=False, without595ace4eb84dd840f - Missing
    bzip2/1.0.8:5be2b7a2110ec8acdbf9a1cea9de5d60747edb34 - Cache
    innoextract/1.9.0@kam/testing:0a420ff5c47119e668867cdb51baff0eca1fdb68 - Cache
    libiconv/1.16:6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7 - Cache
    xz_utils/5.2.5:6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7 - Cache
    zlib/1.2.11:6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7 - Cache
```

...how do I configure the package so that if it's in the Cache, it doesn't have to resolve its requirements? Shouldn't it be possible to get an _already built_ tool without having to build the _static libraries_ that tool uses? What am I missing?

Reply:
> I'll try taking it the private keyword, then. If it really causes problems for us, I'll put in our fork.

Wow, what did I write? What I'm saying is that if I need to tweak the recipe for our purposes, I'll carry that tweak in our fork.